src/App.tsx

```diff
@@ -1,26 +1,219 @@
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { USER_ID } from './api/todos';
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+import * as todosFromServer from './api/todos';
+import { wait } from './utils/fetchClient';
+import { TodoForm } from './components/TodoForm/TodoForm';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFooter } from './components/TodoFooter/TodoFooter';
+import { Status } from './types/Status';
+import { TodoItem } from './components/TodoItem/TodoItem';
 
-const USER_ID = 0;
+const getTodosByStatus = (status: string, todos: Todo[]) => {
+  const preperedTodos = [...todos];
+
+  if (status) {
+    switch (status) {
+      case Object.keys(Status)[Object.values(Status).indexOf(Status.active)]:
+        return preperedTodos.filter(todo => !todo.completed);
+      case Object.keys(Status)[Object.values(Status).indexOf(Status.completed)]:
+        return preperedTodos.filter(todo => todo.completed);
+      default:
+        return preperedTodos;
+    }
+  }
+
+  return preperedTodos;
+};
 
 export const App: React.FC = () => {
+  // const [titleError, setTitleError] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  // const [loadError, setLoadError] = useState(false);
+  // const [addError, setAddError] = useState(false);
+  // const [deleteError, setDeleteError] = useState(false);
+  // const [updateError, setUpdateError] = useState(false);
+  const [status, setStatus] = useState('all');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodos, setLoadingTodos] = useState<Todo[]>([]);
+  const [edit, setEdit] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const temp = (currentTodo: Todo) => {
+    setLoadingTodos(prevArray => [...prevArray, currentTodo]);
+  };
+
+  const filteredTodos = getTodosByStatus(status, todos);
+
+  async function addTodo(newTodoTitle: string) {
+    const editedTitle = newTodoTitle.trim();
+
+    if (!editedTitle) {
+      setErrorMessage('Title should not be empty');
+      wait(3000).then(() => setErrorMessage('Title should not be empty'));
+
+      return;
+    } else {
+      setTempTodo({
+        id: 0,
+        userId: 839,
+        title: editedTitle,
+        completed: false,
+      });
+
+      return todosFromServer
+        .createTodos({
+          userId: 839,
+          title: editedTitle,
+          completed: false,
+        })
+        .then(newTodo => {
+          setTodos(prevTodos => [...prevTodos, newTodo]);
+          setTempTodo(null);
+        })
+        .catch(error => {
+          setErrorMessage('Unable to add a todo');
+          setTempTodo(null);
+          wait(3000).then(() => setErrorMessage(''));
+          throw error;
+        });
+    }
+  }
+
+  async function updateTodo(
+    updatedTodo: Todo,
+    // successUpdateState?: VoidFunction,
+  ): Promise<void> {
```

Comment:
It's important to use one way of creating functions, fix it everywhere
```suggestion
  const updateTodo = async (
    updatedTodo: Todo,
    // successUpdateState?: VoidFunction,
  ): Promise<void> => {
```